rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -473,16 +533,41 @@ extern "C" rmw_ret_t rmw_service_set_listener_callback(
   const void * user_data,
   const void * service_handle)
 {
-  (void)rmw_service;
-  (void)callback;
-  (void)user_data;
-  (void)service_handle;
-  // auto service = static_cast<CddsService *>(rmw_service->data);
-  // service->setCallback(user_data, callback, service_handle);
-  RCUTILS_LOG_ERROR_NAMED(
-    "rmw_node.cpp",
-    "rmw_service_set_listener_callback: not supported (yet)");
-  return RMW_RET_UNSUPPORTED;
+  auto srv = static_cast<CddsService *>(rmw_service->data);
+  auto entity_to_listen = srv->service.sub->enth;
```

Comment:
`auto` should be used when the dataype can be understood from the context, which is not the case here

Reply:
done in 010fb92